fml/message_loop_task_queues.cc

```diff
@@ -336,32 +358,28 @@ TaskSource::TopTask MessageLoopTaskQueues::PeekNextTaskUnlocked(
     TaskQueueId owner) const {
   FML_DCHECK(HasPendingTasksUnlocked(owner));
   const auto& entry = queue_entries_.at(owner);
-  const TaskQueueId subsumed = entry->owner_of;
-  if (subsumed == _kUnmerged) {
+  if (entry->owner_of.empty()) {
     return entry->task_source->Top();
   }
 
   TaskSource* owner_tasks = entry->task_source.get();
-  TaskSource* subsumed_tasks = queue_entries_.at(subsumed)->task_source.get();
-
-  // we are owning another task queue
-  const bool subsumed_has_task = !subsumed_tasks->IsEmpty();
-  const bool owner_has_task = !owner_tasks->IsEmpty();
-  fml::TaskQueueId top_queue_id = owner;
-  if (owner_has_task && subsumed_has_task) {
-    const auto owner_task = owner_tasks->Top();
-    const auto subsumed_task = subsumed_tasks->Top();
-    if (owner_task.task > subsumed_task.task) {
-      top_queue_id = subsumed;
-    } else {
-      top_queue_id = owner;
+  std::vector<TaskSource::TopTask> candidate_top_tasks;
```

Comment:
It would be better to make this a `vector<TaskSource::TopTask&>` to avoid copying TopTask's.  Since it just lives for this invocation that should be fine.

Reply:
@gaaclarke  I tried some ways to make it as a `TaskSource::TopTask&` reference, but `owner_tasks->Top()` passing a value of `struct TopTask`, Where can is store the values of reference of every item of owner_of's `tasks->Top()` value?

Reply:
I don't think you need a vector here to accumulate the tasks, you could just store the current `min_task` and compare against all the subsumed tasks.

Reply:
Thanks for this suggestion! I change it to `std::optional<TopTask>`, bacause `task_source->Top()` return a value of TopTask, `optional` can provide the memory space for the top task.